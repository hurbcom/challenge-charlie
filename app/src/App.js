/* eslint-disable no-unused-vars */
import React from 'react';
import Background from './components/Background.js';
import Search from './components/Search.js';

import './style/globalStyle.css';

function App() {
  return (
    <>
      <div className='content-div'>
        <Search />
      </div>
      <Background />;
    </>
  );
}

export default App;
