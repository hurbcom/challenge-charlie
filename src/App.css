:root {
  --gray-400: #9ca3af;
  --gray-500: #6b7280;
}

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

html, body, input {
  font-family: 'Nunito', sans-serif;
  font-weight: 300;
}

.app__container {
  width: 100vw;
  height: 100vh;
  overflow: hidden;
}
