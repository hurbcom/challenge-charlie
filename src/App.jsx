export default () => {
    return <h1>LETS SEE THE WEATHER</h1>;
};
