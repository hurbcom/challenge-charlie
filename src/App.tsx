
function App() {
  return (
    <div className="App">
      hello world
    </div>
  );
}

export default App;
