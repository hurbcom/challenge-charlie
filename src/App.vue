<template>
  <div class="wrapper-app" :style="backgroundProps">
    <div class="container d-flex flex-column">
        <Search />

        <div class="wrapper-weather d-flex flex-column">
            <div class="d-flex today-weather">
                <div class="icon icon-weather">
                    <span class="icon" data-icon="B"></span>
                </div>

                <div class="details-weather d-flex flex-column">
                    <p class="date">Hoje</p>
                    <p class="temperature">32º</p>

                    <p class="state">Ensolarado</p>

                    <p class="wind">Vento: NO 6.4km/h</p>
                    <p class="humidty">Humidade: 78%</p>
                    <p class="pressure">Pressão: 1003hPa</p>
                </div>
            </div>

            <div class="d-flex tomorrow-weather">
                <div class="icon icon-weather"></div>
                <div class="details-weather d-flex flex-column">
                    <p class="date">Amanhã</p>
                    <p class="temperature">25º</p>
                </div>
            </div>

            <div class="d-flex after-tomorrow-weather">
                <div class="icon icon-weather"></div>
                <div class="details-weather d-flex flex-column">
                    <p class="date">Depois de Amanhã</p>
                    <p class="temperature">22º</p>
                </div>
            </div>
        </div>
    </div>
  </div>
</template>

<script>
import axios from 'axios';

import Search from './components/Search.vue';

export default {
    name: 'App',
    components: {
        Search,
    },
    data: () => ({
        baseApiImageBackground: 'http://www.bing.com',
        urlForBackgroundImage: '',
    }),
    methods: {
        async getImageBackground() {
            const apiUrl = `${this.baseApiImageBackground}/HPImageArchive.aspx?format=js&idx=0&n=1&mkt=pt-BR`;
            const { data: { images } } = await axios.get(apiUrl);
            const { url } = images[0];
            this.urlForBackgroundImage = `${this.baseApiImageBackground}${url}`;
        }
    },
    computed: {
        backgroundProps() {
            return {'background-image': `url(${this.urlForBackgroundImage})`}
        }
    },
    created() {
        this.getImageBackground();
    }
};
</script>

<style lang="scss">
    @import url('./fonts/stylesheet.css');
    @import url('./sass/icons.scss');

    * {
        margin: 0;
        padding: 0;
        box-sizing: border-box;
        font-family: 'Roboto', sans-serif;
    }

    html, body {
        background-color: #cdcdcd;
    }

    .d-flex {
        display: flex;

        &.flex-column {
            flex-direction: column;
        }
    }

    .wrapper-app {
        .container {
            height: 100vh;
            width: 50vw;
            min-width: 505px;
            margin: 0 auto;
            background-color: #fff;

            .wrapper-weather {
                > div {
                    padding: 10px 0 20px 0;
                    height: 20vh;

                    &:first-child {
                        height: 45vh;
                    }
                }

                .icon-weather {
                    width: 60%;
                    text-align: center;

                    .icon::before {
                        font-size: 15rem;
                        color: #fff;
                    }
                }

                .details-weather {
                    p {
                        color: #fff;
                        font-size: 1.3rem;
                        font-weight: 400;

                        &.date {
                            text-transform: uppercase;
                        }

                        &.state {
                            padding: 15px 0 20px;
                        }

                        &.wind,
                        &.humidty,
                        &.pressure {
                            font-size: 1rem;
                        }
                    }
                }

                .today-weather { background-color:rgb(212, 198, 0); }
                .tomorrow-weather { background-color:rgb(235, 223, 59); }
                .after-tomorrow-weather { background-color:rgb(165, 156, 26); }
            }
        }
    }
</style>