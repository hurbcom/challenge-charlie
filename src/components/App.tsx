import React from "react";
import Background from "./Background/Background";

const App = () => {
  return (
    <>
      <Background />
    </>
  );
};

export default App;
