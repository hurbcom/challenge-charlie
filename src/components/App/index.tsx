import { useEffect } from "react";

import styled from "styled-components";
import useBingWallpapper from "hooks/useBingWallpapper";
import Weather from "components/Weather";

const AppBase = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: 100vh;
`;

const Copyright = styled.div`
    font-size: 9px;
    font-weight: 200;
    width: 50%;
    padding: 4px;
`;

const App = () => {
    const { loadWallpaperOfDay, wallpapper } = useBingWallpapper();

    useEffect(() => {
        if (!!wallpapper?.url) {
            document.body.style.setProperty(
                "--body-background-image",
                `url(${wallpapper.url})`
            );
        }
    }, [wallpapper]);

    useEffect(() => {
        loadWallpaperOfDay();
    }, []);

    return (
        <>
            <AppBase>
                <Weather />
                {!!wallpapper?.copyright ? (
                    <Copyright>{wallpapper?.copyright}</Copyright>
                ) : null}
            </AppBase>
        </>
    );
};

export default App;
