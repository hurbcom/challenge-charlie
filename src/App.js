import { Background } from "./components/background";
import GlobalStyles from "./global-styles";
import "./font/stylesheet.css";
import { MainComponent } from "./components/main";

function App() {
    return (
        <div className="App">
            <GlobalStyles />
            <Background>
                <MainComponent />
            </Background>
        </div>
    );
}

export default App;
