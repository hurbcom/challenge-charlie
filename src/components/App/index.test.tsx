import { render, screen } from "@testing-library/react";
import App from ".";

const image = {
    url: "/th?id=OHR.KiteDay_PT-BR2236604621_1920x1080.jpg&rf=LaDigue_1920x1080.jpg&pid=hp",
    copyrightlink:
        "https://www.bing.com/search?q=soltar+pipas&form=hpcapt&mkt=pt-br",
    copyright:
        "Adelaide Dia Internacional de Solte sua Pipa, na Australia (Â© Andrey Moisseyev/Alamy)",
};

jest.mock("../../hooks/useBingWallpapper", () => {
    return jest
        .fn()
        .mockImplementationOnce(() => ({
            wallpapper: {},
            loadWallpaperOfDay: jest.fn().mockImplementationOnce(() =>
                Promise.resolve({
                    json: () =>
                        Promise.resolve({
                            images: [image],
                        }),
                })
            ),
        }))
        .mockImplementationOnce(() => ({
            wallpapper: {
                ...image,
            },
            loadWallpaperOfDay: jest.fn().mockImplementationOnce(() =>
                Promise.resolve({
                    json: () =>
                        Promise.resolve({
                            images: [image],
                        }),
                })
            ),
        }));
});

describe("<App />", () => {
    test("renders the <App/> component without background", () => {
        const { asFragment } = render(<App />);
        expect(asFragment).toMatchSnapshot();
    });

    test("renders the <App/> component with background", () => {
        const { asFragment } = render(<App />);
        expect(asFragment).toMatchSnapshot();
    });
});
