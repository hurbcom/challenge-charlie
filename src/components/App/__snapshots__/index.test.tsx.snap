// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders the <App/> component with background 1`] = `[Function]`;

exports[`<App /> renders the <App/> component without background 1`] = `[Function]`;
